### PR TITLE
Diagram support for watchdogs

### PR DESCRIPTION
**Pull request by edwardalee**
This PR adds diagram support for watchdogs. They are shown as squares with a "W".

### CODE REVIEW

File: core/src/main/java/org/lflang/diagram/synthesis/LinguaFrancaSynthesis.java [hunk 1]
```diff
@@ -1251,6 +1258,33 @@ private Collection<KNode> transformReactorNetwork(
...
+      }
+      // connect source
+      for (KPort source : watchdogSources.get(watchdog)) {
+        connect(createDelayEdge(watchdog), source, ports.getKey());
+      }
+
+      // connect targets
+      for (KPort target : watchdogDestinations.get(watchdog)) {
+        connect(createDelayEdge(watchdog), ports.getValue(), target);
+      }
```
Comment (soerendomroes) [inline core/src/main/java/org/lflang/diagram/synthesis/LinguaFrancaSynthesis.java:1285]: You might want to enable layout annotation for watchdogs, as done [here](https://github.com/lf-lang/lingua-franca/blob/4bb000e4dc88988e40e681592a2fc16f89908a2b/core/src/main/java/org/lflang/diagram/synthesis/LinguaFrancaSynthesis.java#L1058) for timers.
The layout annotations should be applied after everything else that sets layout options finishes so after `addWatchdogFigureAndPorts` was called.

Adding a method to configure layout for watchdogs in LayoutPostProcessing, as done [here](https://github.com/lf-lang/lingua-franca/blob/4bb000e4dc88988e40e681592a2fc16f89908a2b/core/src/main/java/org/lflang/diagram/synthesis/LinguaFrancaSynthesis.java#L1057) might not be necessary and I will add this once a new ELK release may add more interesting ordering related layout options.

File: core/src/main/java/org/lflang/diagram/synthesis/styles/LinguaFrancaShapeExtensions.java [hunk 2]
```diff
@@ -839,53 +879,6 @@ public KText addTextButton(KContainerRendering container, String text) {
     return textToAdd;
   }
 
-  /** Creates the triangular line decorator with text. */
-  public KPolygon addActionDecorator(KPolyline line, String text) {
```
Comment (soerendomroes) [inline core/src/main/java/org/lflang/diagram/synthesis/styles/LinguaFrancaShapeExtensions.java:843]: I suppose this might no longer be necessary? If this is the case, please resolve this comment.

File: core/src/main/java/org/lflang/diagram/synthesis/LinguaFrancaSynthesis.java (hunk 1 above)
Comment (edwardalee) [reply core/src/main/java/org/lflang/diagram/synthesis/LinguaFrancaSynthesis.java]: Done in 521a50245

File: core/src/main/java/org/lflang/diagram/synthesis/styles/LinguaFrancaShapeExtensions.java (hunk 2 above)
Comment (edwardalee) [reply core/src/main/java/org/lflang/diagram/synthesis/styles/LinguaFrancaShapeExtensions.java:843]: This was not called anywhere, so I deleted it.

**Review by soerendomroes — APPROVED** [review]

File: core/src/main/java/org/lflang/LinguaFranca.xtext [hunk 3]
```diff
@@ -218,6 +218,7 @@ Deadline:
     'deadline' '(' delay=Expression ')' code=Code;
 
 Watchdog:
+    (attributes+=Attribute)*
```
Comment (lhstrh) [inline core/src/main/java/org/lflang/LinguaFranca.xtext:221]: I'm surprised to see a grammar change in a PR about diagram support. Why this change?

Comment (edwardalee) [reply core/src/main/java/org/lflang/LinguaFranca.xtext:221]: So that you can add annotations, like labels and layout annotations.

Comment (lhstrh) [reply core/src/main/java/org/lflang/LinguaFranca.xtext:221]: :+1:

**Review by lhstrh — COMMENTED** [review]
Should this PR include an update to `reactor-c`?

**Comment by soerendomroes** [comment]
The watchdog seems to be missing [here](https://github.com/lf-lang/lingua-franca/blob/4865086bf50862f82fe5d476471688cc5fbe59f1/core/src/main/java/org/lflang/AttributeUtils.java#L67-L86) to be able to get its attributes.
